src/components/form/Form.js

```diff
@@ -0,0 +1,215 @@
+/* eslint-disable no-unused-expressions */
+/* eslint-disable no-use-before-define */
+/* eslint-disable jsx-a11y/no-static-element-interactions */
+/* eslint-disable jsx-a11y/click-events-have-key-events */
+/* eslint-disable react/destructuring-assignment */
+/* eslint-disable react/prop-types */
+import React from 'react';
+import './Form.css';
+
+function Form(props) {
+  const [displayForm, setDisplayForm] = React.useState(true);
+  const [isCompleted, setIsCompleted] = React.useState(false);
+
+  const initialValues = { company: '', email: '', description: '', name: '' };
+  const [inputValues, setInputValues] = React.useState(initialValues);
+  const [inputErrors, setInputErrors] = React.useState({});
+  const [onSubmitClick, setOnSubmitClick] = React.useState(false);
+
+  // Input change handler
+  const handleChange = (e) => {
+    const { name, value } = e.target;
+    setInputValues({ ...inputValues, [name]: value });
+  };
+
+  // This will clear all input values and error messages when user closes form without submitting.
+  // Lots of repeated code!! How to improve this?
+  function clearValues() {
+    inputValues.company = '';
+    inputValues.email = '';
+    inputValues.description = '';
+    inputValues.name = '';
+    inputErrors.company = '';
+    inputErrors.email = '';
+    inputErrors.description = '';
+    inputErrors.name = '';
+  }
+
+  // Combines closing modal and clearing of inputs and error messages
+  function closeModal() {
+    props.onClose();
+    setTimeout(clearValues, 1000);
+    setTimeout(() => {
+      // Display form instead of confirmation message when reopening popup
+      setDisplayForm(true);
+      setIsCompleted(false);
+    }, 1000);
+    window.removeEventListener('keydown', handleEscClose);
+  }
+
+  function handleModalClick(e) {
+    if (e.target.classList.contains('modal_open')) {
+      closeModal();
+    }
+  }
+
+  function handleEscClose(e) {
+    if (e.key === 'Escape') {
+      closeModal();
+    }
+  }
+  window.addEventListener('keydown', handleEscClose);
+
+  // When user clicks on submit button, form will be validated
+  function handleSubmit(e) {
+    e.preventDefault();
+    setInputErrors(validate(inputValues));
```

Comment:
fresh! innovative! i would never have though of this! RuPaul approves!

Reply:
Yay! 🎉